### PR TITLE
Refactor/card pair session key loading

### PR DESCRIPTION
**Pull request by amanCypherock**
The intention of this change is to include the pairing check in the `card_initialize_applet` API.
This way session keys are loaded by this API when the secret channel is required and returns `CARD_OPERATION_PAIRING_REQUIRED` in case of an unpaired card.

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [PRF-4795 Card operation - read share from card for reconstruction](https://app.clickup.com/t/9002019994/PRF-4795)

File: src/level_four/tap_cards/controller/controller_tap_cards.h [hunk 1]
```diff
@@ -44,6 +44,7 @@ typedef struct NFC_connection_data {
   uint8_t card_key_id[4];
   uint8_t recovery_mode;
   uint8_t card_absent_retries;
+  bool pairing_required;
```
Comment (ujjwal-cyph) [inline src/level_four/tap_cards/controller/controller_tap_cards.h:47]: suggestion to rename `pairing_required` to `init_session_keys`

Comment (amanCypherock) [reply src/level_four/tap_cards/controller/controller_tap_cards.h]: updated

**Review by ujjwal-cyph — APPROVED** [review]

File: src/card_operations/card_internal.c [hunk 2]
```diff
@@ -256,6 +282,15 @@ card_error_type_e card_initialize_applet(card_operation_data_t *card_data) {
           NFC_RETURN_ABORT_ERROR(card_data,
                                  ui_critical_card_health_migrate_data);
         }
+
+        /* Check if pairing is required */
+        if (true == card_data->nfc_data.init_session_keys) {
+          /* Return pairing error if not paired */
+          if (false == load_card_session_key(card_data->nfc_data.card_key_id)) {
+            NFC_RETURN_ERROR_TYPE(card_data, CARD_OPERATION_PAIRING_REQUIRED);
```
[the remark this answers is not in the document]
Comment (amanCypherock) [reply src/card_operations/card_internal.c:290]: added

**Pull request merged**